test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,379 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using static Microsoft.Azure.WebJobs.Host.EndToEndTests.Scale.ScaleHostEndToEndTests;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.Scale
+{
+    [Trait(TestTraits.CategoryTraitName, TestTraits.ScaleMonitoring)]
+    public class ScaleHostEndToEndTests
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+        private const string Function3Name = "Function3";
+
+        private const string ATriggerType = "testExtensionATrigger";
+        private const string BTriggerType = "testExtensionBTrigger";
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabled)
+        {
+            var triggerMetadata = new List<TriggerMetadata>()
+            {
+                new TriggerMetadata(new JObject { { "functionName", $"{Function1Name}" }, { "type", $"{ATriggerType}" } }),
+                new TriggerMetadata(new JObject { { "functionName", $"{Function2Name}" }, { "type", $"{ATriggerType}" } }),
+                new TriggerMetadata(new JObject { { "functionName", $"{Function3Name}" }, { "type", $"{BTriggerType}" } }, new Dictionary<string, object> { { "foo", "bar" } })
+            };
+
+            string hostJson =
+            @"{
+              ""extensions"": {
+                ""testExtensionA"" : {
+                  ""foo"": 16
+                },
+                ""testExtensionB"" : {
+                  ""bar"": 8
+                }
+              }
+            }";
+
+            string hostId = "test-host";
+            var loggerProvider = new TestLoggerProvider();
+
+            IHostBuilder hostBuilder = new HostBuilder();
+            hostBuilder.ConfigureLogging(configure =>
+            {
+                configure.SetMinimumLevel(LogLevel.Debug);
+                configure.AddProvider(loggerProvider);
+            });
+            hostBuilder.ConfigureAppConfiguration((hostBuilderContext, config) =>
+            {
+                // Adding host.json here
+                config.AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(hostJson)));
+
+                // Adding app settings
+                config.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    { "app_setting1", "value1" },
+                    { "app_setting2", "value2"},
+                    { $"{Constants.HostingConfigSectionName}:Microsoft.Azure.WebJobs.Host.EndToEndTests", "1" }
```

Comment:
We should also test the case where the app setting is defined with double underscore __ instead of colon :

Reply:
I think  "__"  shoud be normalized to ":" by ScalerHost creator (SC).